packages/opc-footer/README.md

```diff
@@ -1,112 +1,138 @@
+* The links needs to have following data type which is defined in `./types.d.ts` file. Below is the mock data example.
+* use the `opcLinkCatagories` setter function to add links.
+```js
+const links = [
+  { "group":"Group name 1",
+    "links":[
+      { "name":"Lorem ipsum", "path":"https://example.com" },
+      { "name":"Lorem ipsum" },
+    ]},
+  { "group":"Group name 2",
+    "links":[
+      { "name":"Lorem ipsum", "path":"https://example.com" },
+      { "name":"Lorem ipsum" },
+      { "name":"Lorem ipsum" },
+  ]},
+];
 
-```sh
-npm install
+const myFooterEl = document.querySelector('opc-footer');
+myFooterEl.opcLinkCatagories = links;
 ```
 
-## Usage
+* The Links with no path fires the custom event called `opc-footer-link:click`, so the user can add custom behavior. for example 
+```js
+  const myFooterElement = document.querySelector('opc-footer');
+  myFooterElement.addEventListener('opc-footer-link:click', (e) => {
+    console.log(e);
+    /// custom behaviors.
+  });
+```
 
-the opc-footer have supports dark theme, by default the light theme is selected. for example
-footer with default theme:
+Default with light theme: 
 
 ```html
 <opc-footer></opc-footer>
 ```
 
-![Demo Footer light with no links](demo/images/Footer-light-no-links.png)
+Dark theme:
+in `html`
+```html
+<opc-footer theme="dark"></opc-footer>
+```
+as well you can change the theme in `js` too by selecting the element reference.
+```js
+const myFooterEl = document.querySelector('opc-footer');
+myFooterEl.theme = 'dark';
+```
 
-here's another with dark
+# Slot
+The `opc-footer` component supports the slot for foot note which is reserved for copyright text. By default it shows the text as Copyright **All Rights Reserved.**
+if you want to change use the slot name as `copyright` for example.
 
 ```html
-<opc-footer theme="dark"></opc-footer>
+<opc-footer>
+  <span slot="copyright">My Company Rights Reserved</span>
+</opc-footer>
 ```
 
-![Demo Footer dark no links](demo/images/Footer-dark-no-links.png)
+# Attributes
 
-To add the links it has a attribute called opc-links-groups. this supports the Array folloing is the structure.
+### `theme`
 
-```js
-[
-  {
-    name: "Example Group Link",
-    links: [
-      {
-        name: "Crunch time",
-        path: "https://example.com",
-      },
-      {
-        name: "Lorem ipsum Link",
-      },
-    ],
-  },
-  ...
-];
-```
-For a demo example see the
-![Demo Footer light with no links](demo/images/Footer-light-no-links.png)
-![Demo Footer dark with links](demo/images/Footer-dark-with-links.png)
+| Theme name |  Value  |
+| ----------------- | ------- | 
+| `light` |  #FFF |
+| `dark` |  #000 |
+
+### `link-catagories`
 
-The Links with no path fires the custom event called footer-link-click, so the user can add
-custom behaviors.
+the better way to add the links is via js API but if you want to implementing `data goes down and events go up` pattern in your code then you can use the following attribute.
```

Comment:
it should change to `We recommend to provide the links via js API but if you want to implement `data goes down and events go up` pattern then you can use the `link-categpries` attribute.